exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/efgs/FederationBatchSigner.java

```diff
@@ -0,0 +1,239 @@
+package fi.thl.covid19.exposurenotification.efgs;
+
+import com.google.protobuf.ByteString;
+import com.google.protobuf.ProtocolStringList;
+import fi.thl.covid19.proto.EfgsProto;
+import org.bouncycastle.asn1.x500.X500Name;
```

Comment:
I have used logic from efgs and it's uses bouncy castle. So, it's easier to check that our signing is compatible with efgs and also there is some more intuitive top-level methods in bouncy castle. Anyway, it could be done with java also.